File: frontend/src/types/favorite.ts
// 收藏相关类型

export interface FavoriteFolder {
  id: number;
  user_id: number;
  name: string;
  is_public: boolean;
  favorites_count: number;
  created_at: string;
}

export interface FavoriteFolderCreate {
  name: string;
  is_public?: boolean;
}

export interface FavoriteFolderUpdate {
  name?: string;
  is_public?: boolean;
}

export interface FavoriteFolderListResponse {
  folders: FavoriteFolder[];
}

export interface FavoriteCreate {
  folder_id: number;
}

export interface FavoriteDelete {
  folder_id?: number;
}

export interface FavoriteResponse {
  favorited: boolean;
  favorites_count: number;
}

export interface FavoriteFolderInfo {
  id: number;
  name: string;
}

export interface FavoriteStatusResponse {
  is_favorited: boolean;
  folders: FavoriteFolderInfo[];
  favorites_count: number;
}

File: backend/schemas/favorite.py
from pydantic import BaseModel, Field


class FavoriteCreate(BaseModel):
    """收藏博客"""
    folder_id: int = Field(..., description="目标文件夹 ID")


class FavoriteDelete(BaseModel):
    """取消收藏（可选指定文件夹）"""
    folder_id: int | None = Field(None, description="指定文件夹 ID（不提供则从所有文件夹移除）")


class FavoriteResponse(BaseModel):
    """收藏/取消收藏响应"""
    favorited: bool
    favorites_count: int


class FavoriteFolderInfo(BaseModel):
    """收藏文件夹信息"""
    id: int
    name: str


class FavoriteStatusResponse(BaseModel):
    """收藏状态响应"""
    is_favorited: bool
    folders: list[FavoriteFolderInfo]
    favorites_count: int
